### PR TITLE
Indexes changes

### PR DESCRIPTION
**Pull request by brunoocasali**
- Add `IndexesQuery` type to handle the indexes endpoints parameters.
- Add `IndexesResults` type to handle the object response in the `GET /indexes` response.
- Change responses in the create/update/delete actions for documents to return `TaskInfo`.

### CODE REVIEW

File: Sources/MeiliSearch/Indexes.swift [hunk 1]
```diff
@@ -101,15 +101,21 @@ public struct Indexes {
    completes the query request. It returns a `Result` object that contains `[Index]`
    value if the request was successful or `Error` if a failure occurred.
    */
-  public static func getAll(config: Config, _ completion: @escaping (Result<[Index], Swift.Error>) -> Void) {
-    Request(config).get(api: "/indexes") { result in
+  public static func getAll(config: Config, params: IndexesQuery? = nil, _ completion: @escaping (Result<Results<Index>, Swift.Error>) -> Void) {
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Indexes.swift:104]: Are we keeping `Result<Index>` and not `IndexResults` (like `DocumentsResult`  until the `GET /indexes` route has specific response fields that differ from other routes?

File: Sources/MeiliSearch/Indexes.swift [hunk 2]
```diff
@@ -346,9 +352,10 @@ public struct Indexes {
    */
   public func getDocument<T>(
     _ identifier: String,
+    fields: [String]? = nil,
     _ completion: @escaping (Result<T, Swift.Error>) -> Void)
   where T: Codable, T: Equatable {
-    self.documents.get(self.uid, identifier, completion)
+    self.documents.get(self.uid, identifier, fields: fields, completion)
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Indexes.swift:358]: Should this not be added in the `Documents` PR ? (I can close my eyes if you want 😂)

File: Sources/MeiliSearch/Indexes.swift [hunk 3]
```diff
@@ -376,10 +383,10 @@ public struct Indexes {
    failure occured.
    */
   public func getDocuments<T>(
-    options: GetParameters? = nil,
-    _ completion: @escaping (Result<[T], Swift.Error>) -> Void)
+    params: DocumentsQuery? = nil,
+    _ completion: @escaping (Result<DocumentsResults<T>, Swift.Error>) -> Void)
   where T: Codable, T: Equatable {
-    self.documents.getAll(self.uid, options, completion)
+    self.documents.getAll(self.uid, params: params, completion)
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Indexes.swift:389]: Same as above

File: Sources/MeiliSearch/Indexes.swift [hunk 4]
```diff
@@ -392,7 +399,7 @@ public struct Indexes {
    */
   public func deleteDocument(
     _ documentId: String,
-    _ completion: @escaping (Result<Task, Swift.Error>) -> Void) {
+    _ completion: @escaping (Result<TaskInfo, Swift.Error>) -> Void) {
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Indexes.swift:402]: Same as above, should this not be added in the `Tasks` PR ?

File: Sources/MeiliSearch/Indexes.swift [hunk 5]
```diff
@@ -560,7 +568,7 @@ public struct Indexes {
    value. If the request was sucessful or `Error` if a failure occured.
    */
   public func resetSynonyms(
-    _ completion: @escaping (Result<Task, Swift.Error>) -> Void) {
+    _ completion: @escaping (Result<TaskInfo, Swift.Error>) -> Void) {
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Indexes.swift:571]: All the `Task` to `TaskInfo` changes in the completion parameter are already changed in this pr #316

File: Sources/MeiliSearch/Indexes.swift [hunk 6]
```diff
@@ -881,22 +889,6 @@ public struct Indexes {
...
-      let rawIndexes: [Index] = try Constants.customJSONDecoder.decode([Index].self, from: data)
-      var indexes: [Index] = []
-      for rawIndex in rawIndexes {
-        indexes.append(Index(uid: rawIndex.uid, primaryKey: rawIndex.primaryKey, createdAt: rawIndex.createdAt, updatedAt: rawIndex.updatedAt))
-      }
-      completion(.success(indexes))
-    } catch {
-      completion(.failure(error))
-    }
-  }
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Indexes.swift:899]: bye bye ❤️

File: Sources/MeiliSearch/Model/IndexesResults.swift [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+/**
+ `IndexesResults` is a wrapper used in the indexes routes to handle the returned data.
+ */
+
+public struct IndexesResults<T: Decodable & Encodable & Equatable>: Codable, Equatable {
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Model/IndexesResults.swift:7]: I see that there is a `IndexesResults` but somewhere above you used `Results<Index>`

File: Sources/MeiliSearch/Indexes.swift (hunk 2 above)
Comment (brunoocasali) [reply Sources/MeiliSearch/Indexes.swift:358]: dammit, I forgot that yes close your eyes ahhaha!

File: Sources/MeiliSearch/Indexes.swift (hunk 4 above)
Comment (brunoocasali) [reply Sources/MeiliSearch/Indexes.swift:402]: Not really. These small chunks are very time-consuming to create a PR with only these changes. Sorry about that!

File: Sources/MeiliSearch/Model/IndexesResults.swift (hunk 7 above)
Comment (brunoocasali) [reply Sources/MeiliSearch/Model/IndexesResults.swift:7]: Nice catch, I totally forgot them! :P

**Review by bidoubiwa — APPROVED** [review]
LGTM 🔥

**Pull request merged**